iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateCouplingDevice.java

```diff
@@ -109,51 +106,112 @@ public void apply(Network network, boolean throwException, ComputationManager co
+
+            nbOpenDisconnectors += bbsList2.size() - 1;
+        } else {
+            createDisconnectorTopology(voltageLevel2, breakerNode2, switchPrefixId, List.of(bbs2), bbs2);
+            LOGGER.warn("No busbar section position extension found on {}, only one disconnector is created.", bbs2.getId());
+            noBusbarSectionPositionExtensionReport(reporter, bbs2);
+        }
+
+        if (nbOpenDisconnectors > 0) {
+            LOGGER.info("{} open disconnectors created on parallel busbar section in voltage level {}", nbOpenDisconnectors, voltageLevel1.getId());
+            openDisconnectorsAddedReport(reporter, voltageLevel1.getId(), nbOpenDisconnectors);
+        }
+    }
+
+    /**
+     * Check if the first side is on the first bar or the second side on the last bar
+     * @param voltageLevel Voltage level in which the busbar sections are located
+     * @param position1 Position on the first side
+     * @param position2 Position on the second side
+     * @return True if the first side is on the first bar or the second side is on the last bar, else False
+     */
+    private boolean checkSides(VoltageLevel voltageLevel, BusbarSectionPosition position1, BusbarSectionPosition position2) {
+        // Check if the first side is on the first bar or the second side on the last bar
+        OptionalInt minBbsIndex = getParallelBusbarSections(voltageLevel, position1).stream()
+            .mapToInt(b -> b.getExtension(BusbarSectionPosition.class).getBusbarIndex()).min();
+        OptionalInt maxBbsIndex = getParallelBusbarSections(voltageLevel, position2).stream()
+            .mapToInt(b -> b.getExtension(BusbarSectionPosition.class).getBusbarIndex()).max();
+        return position1.getBusbarIndex() == minBbsIndex.orElseThrow()
+            || position2.getBusbarIndex() == maxBbsIndex.orElseThrow();
+    }
+
+    /**
+     * Computes the list of the bars on which to connect the coupling device for the current side and creates the
+     * disconnectors for the different parallel busbar sections.
+     * @param voltageLevel Voltage level in which the busbar sections are located
+     * @param position Position of the current bar
+     * @param bbs Current bar
+     * @param bbsOnSameSection True if the two busbar sections are located on the same section
+     * @param breakerNode Node on the current site of the breaker
+     * @param avoidLastBar If true, the last bar will not be connected, if false the first one will not be connected.
+     * @return List of the busbar sections on which a connection was made
+     */
+    private List<BusbarSection> computeBbsListAndCreateTopology(VoltageLevel voltageLevel, BusbarSectionPosition position, BusbarSection bbs, boolean bbsOnSameSection, int breakerNode, boolean avoidLastBar) {
+
+        // List of the bars for the second section
```

Comment:
This old comment is not appropriate anymore

Reply:
changed